FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java

```diff
@@ -109,11 +146,32 @@ public void update(Observable arg0, Object argN) {
     }
 
     // Automation Hook: to write Notifications to a temporary file on the device/emulator
-    String text="Notification Received: " + arg1 + " - state: " + player.getState();
+    text="Notification Received: " + arg1 + " - state: " + player.getState();
     // Automation Hook: Write the event text along with event count to log file in sdcard if the log file exists
     Playbacklog.writeToSdcardLog(text);
 
     Log.d(TAG, "Notification Received: " + arg1 + " - state: " + player.getState());
   }
 
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+                                         String permissions[], int[] grantResults) {
+    switch (requestCode) {
+      case MY_PERMISSIONS_REQUEST_READ_CONTACTS: {
+        // If request is cancelled, the result arrays are empty.
+        if (grantResults.length > 0
+                && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+
+          // permission was granted, yay! Do the
+          // contacts-related task you need to do.
+          Playbacklog.writeToSdcardLog(text);
```

Comment:
This is out of context, if you use a boolean flag as I told you before, you would be setting it to `true` here. Then whenever you want to use `Playbacklog.writeToSdcardLog(aString);` you should first verify if you have the permission to write to the SdCard.